Jetcaster/app/src/main/java/com/example/jetcaster/ui/player/PlayerScreen.kt

```diff
@@ -0,0 +1,358 @@
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(vertical = 8.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceEvenly
+    ) {
+        val buttonsModifier = Modifier
+            .size(48.dp)
+            .semantics { role = Role.Button }
+
+        Image(
+            imageVector = Icons.Filled.SkipPrevious,
+            contentDescription = stringResource(R.string.cd_skip_previous),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.tint(LocalContentColor.current),
+            modifier = buttonsModifier
+        )
+        Image(
+            imageVector = Icons.Filled.Replay10,
+            contentDescription = stringResource(R.string.cd_reply10),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.tint(LocalContentColor.current),
+            modifier = buttonsModifier
+        )
+        Image(
+            imageVector = Icons.Rounded.PlayCircleFilled,
+            contentDescription = stringResource(R.string.cd_play),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.tint(LocalContentColor.current),
+            modifier = Modifier
+                .size(72.dp)
+                .semantics { role = Role.Button }
+        )
+        Image(
+            imageVector = Icons.Filled.Forward30,
+            contentDescription = stringResource(R.string.cd_forward30),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.tint(LocalContentColor.current),
+            modifier = buttonsModifier
+        )
+        Image(
+            imageVector = Icons.Filled.SkipNext,
+            contentDescription = stringResource(R.string.cd_skip_next),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.tint(LocalContentColor.current),
+            modifier = buttonsModifier
+        )
+    }
+}
+
+/**
+ * Theme that updates the colors dynamically depending on the podcast image URL
+ */
+@Composable
+private fun PlayerDynamicTheme(
+    uiState: PlayerUiState,
+    content: @Composable () -> Unit
+) {
+    val surfaceColor = MaterialTheme.colors.surface
+    val dominantColorState = rememberDominantColorState(
+        defaultColor = MaterialTheme.colors.surface
+    ) { color ->
+        // We want a color which has sufficient contrast against the surface color
+        color.contrastAgainst(surfaceColor) >= MinContrastOfPrimaryVsSurface
+    }
+    DynamicThemePrimaryColorsFromImage(dominantColorState) {
+        // Update the dominantColorState with colors coming from the podcast image URL
+        LaunchedEffect(uiState.podcastImageUrl) {
+            if (uiState.podcastImageUrl.isNotEmpty()) {
+                dominantColorState.updateColorsFromImageUrl(uiState.podcastImageUrl)
+            } else {
+                dominantColorState.reset()
+            }
+        }
```

Comment:
Very small nit: is there a reason this is inside the `DynamicThemePrimaryColorsFromImage` block?

Reply:
Yeah, this is in charge of updating to the right color. I want this logic to be embedded in here